webapp/channels/src/components/cloud_preview_modal/preview_modal_controller.tsx

```diff
@@ -0,0 +1,137 @@
+            <div className='preview-modal-controller__pagination'>
+                <div
+                    className='preview-modal-controller__pagination-dots'
+                    data-testid='pagination-dots'
+                >
+                    {contentData.map((_, index) => (
+                        <div
+                            key={`dot-${index}`}
+                            className={`preview-modal-controller__dot ${index === currentIndex ? 'preview-modal-controller__dot--active' : ''}`}
+                        />
+                    ))}
+                </div>
+                <span className='preview-modal-controller__page-counter'>
+                    {currentIndex + 1}{'/'}{contentData.length}
+                </span>
+            </div>
+
+            <div className='preview-modal-controller__navigation-buttons'>
+                {isFirstSlide && (
+                    <button
+                        className='preview-modal-controller__skip-button'
+                        onClick={handleSkip}
+                    >
+                        <FormattedMessage
+                            id='cloud_preview_modal.skip'
+                            defaultMessage='Skip for now'
+                        />
+                    </button>
+                )}
+
+                {!isFirstSlide && (
+                    <button
+                        className='preview-modal-controller__nav-button preview-modal-controller__nav-button--tertiary'
+                        onClick={handlePrevious}
+                    >
+                        <ChevronLeftIcon size={18}/>
+                        <FormattedMessage
+                            id='cloud_preview_modal.previous'
+                            defaultMessage='Previous'
+                        />
+                    </button>
+                )}
+
+                <button
+                    className='preview-modal-controller__nav-button preview-modal-controller__nav-button--primary'
+                    onClick={isLastSlide ? onClose : handleNext}
+                >
+                    {isLastSlide ? (
+                        <FormattedMessage
+                            id='cloud_preview_modal.done'
+                            defaultMessage='Finish'
+                        />
+                    ) : (
+                        <>
+                            <FormattedMessage
+                                id='cloud_preview_modal.next'
+                                defaultMessage='Next'
+                            />
+                            <ChevronRightIcon size={18}/>
+                        </>
+                    )}
+                </button>
+            </div>
+        </div>
+    );
+
+    return (
+        <GenericModal
+            show={show}
+            onHide={onClose}
+            compassDesign={true}
+            enforceFocus={true}
+            ariaLabel='Cloud Preview Introduction'
```

Comment:
This needs to be translatable either by translating it here or by referring to another element that contains this text with `aria-labelledby`